### PR TITLE
M1502: Expand HTTP request response monitoring-2

### PR DESCRIPTION
**Pull request by akumar21NCSU**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8845)

### CODE REVIEW

**Comment by jdm** [comment]
Note: you'll need to run `./mach test-tidy` and address the errors it reports.

**Comment by jdm** [comment]
This is great! I'm excited to see the resulting in Firefox!
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 33 unresolved discussions, some commit checks failed.

---

*[components/devtools/actors/network_event.rs, line 15 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jw425lwhJVu_uX0TE:-K4jw425lwhJVu_uX0TF:-1302342800) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L15)):*
nit: remove the spaces after { and before }

---

*[components/devtools/actors/network_event.rs, line 175 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxVA5ki9gd-k4tK0J:-K4jxVA5ki9gd-k4tK0K:816765140) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L175)):*
We can do `for item in &self.request_headers {` instead.

---

*[components/devtools/actors/network_event.rs, line 179 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxfcR7iIIxPcx99cE:-K4jxfcR7iIIxPcx99cF:1697014971) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L179)):*
I think we can use `+=` here instead.

---

*[components/devtools/actors/network_event.rs, line 194 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxwpnIOMq6wSwQGlX:-K4jxwpnIOMq6wSwQGlY:586617401) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L194)):*
Let's call this `cookies` instead.

---

*[components/devtools/actors/network_event.rs, line 196 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxpUwQaxSxzgsJ4Ne:-K4jxpUwQaxSxzgsJ4Nf:928495084) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L196)):*
`for cookie in &req_cookies {`

---

*[components/devtools/actors/network_event.rs, line 226 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jyg7n_7lBSmuvfsXI:-K4jyg7n_7lBSmuvfsXJ:-185109888) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L226)):*
I believe we can collapse these blocks into one by using `postData: self.request.body.clone()`.

---

*[components/devtools/actors/network_event.rs, line 231 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jypq6kSZD2GlLT-qs:-K4jypq6kSZD2GlLT-qt:-731500234) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L231)):*
`if let Some(ref headers) = self.response.headers {`

---

*[components/devtools/actors/network_event.rs, line 237 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jywWfAx64VLmBcHYH:-K4jywWfAx64VLmBcHYI:-1395622715) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L237)):*
`for item in &headers {`

---

*[components/devtools/actors/network_event.rs, line 241 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jz1kLHRI4P4Kiqd4t:-K4jz1kM7BdC8facZ_34:-405773405) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L241)):*
We can probably use +=.

---

*[components/devtools/actors/network_event.rs, line 257 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jz9LppfHOS7dTF2rk:-K4jz9LppfHOS7dTF2rl:-1472251365) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L257)):*
Let's call this `cookies`.

---

*[components/devtools/actors/network_event.rs, line 259 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jz6cpadkSVAlXZDo5:-K4jz6cpadkSVAlXZDo6:-2027813147) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L259)):*
`for cookie in &res_cookies`

---

*[components/devtools/actors/network_event.rs, line 273 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jzDqaKxO9NmA2jkez:-K4jzDqaKxO9NmA2jkf-:-1775418092) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L273)):*
nit: no need for these blank lines.

---

*[components/devtools/actors/network_event.rs, line 276 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jzGhmQupGXTLWuoAq:-K4jzGhmQupGXTLWuoAr:883358301) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L276)):*
We can collapse these with `content: self.response.body.clone()`

---

*[components/devtools/actors/network_event.rs, line 362 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jzghfaVaW8WndyAmA:-K4jzghfaVaW8WndyAmB:-1606335359) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L362)):*
We can use `.strftime("%m/%d/%Y %H:%M:%S").unwrap().to_string()` for that (changing the format string as necessary).

---

*[components/devtools/actors/network_event.rs, line 375 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rry4VrsJAEpo_f4JN:-K4rry4VrsJAEpo_f4JO:-1492696153) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L375)):*
This comment can be removed now.

---

*[components/devtools/actors/network_event.rs, line 377 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4k-Ww7FL6DWuYDLlnV:-K4k-Ww7FL6DWuYDLlnW:-800411758) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L377)):*
Let's remove this.

---

*[components/devtools/actors/network_event.rs, line 379 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jzuw3_NfrdAjvFYi-:-K4jzuw3_NfrdAjvFYi0:1473059758) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L379)):*
This block can become `let hSize = self.response.headers.as_ref().map(|headers| headers.len() as u32);`

---

*[components/devtools/actors/network_event.rs, line 385 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4k-1ak2IKv6nGFeQmt:-K4k-1ak2IKv6nGFeQmu:-1433528948) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L385)):*
This block can become
```rust
let (status_code, status_message) =
    self.response.status.as_ref().map(|RawStatus(code, text)| (code, text.into_owned())).unwrap_or((0, "".to_owned()));
```

---

*[components/devtools/actors/network_event.rs, line 404 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rsTKsiEw5GudYxcFt:-K4rsTKsiEw5GudYxcFu:-731500234) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L404)):*
`if let Some(ref headers) = self.response.headers {`

---

*[components/devtools/actors/network_event.rs, line 406 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rsZlMMU2jzecmZFUK:-K4rsZlMMU2jzecmZFUL:-1018716655) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L406)):*
We can assign mString directly here instead of using the intermediate `mimetype` variable.

---

*[components/devtools/actors/network_event.rs, line 426 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rshqcMb7SKENE90Vz:-K4rshqcMb7SKENE90W-:-731500234) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L426)):*
`if let Some(ref headers) = self.response.headers {`

---

*[components/devtools/actors/network_event.rs, line 428 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rslVrYcmZw_XQbDlE:-K4rslVrYcmZw_XQbDlF:-2094486507) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L428)):*
We can assign cookies_size here directly instead of using the `cookietype` variable.

---

*[components/devtools/actors/network_event.rs, line 443 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rsvunk4srh1a1Rw2f:-K4rsvunk4srh1a1Rw2g:-731500234) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L443)):*
`if let Some(ref headers) = self.response.headers {`

---

*[components/devtools/actors/network_event.rs, line 450 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rtJ1Fl9Vk_-wBlXB4:-K4rtJ1Fl9Vk_-wBlXB5:851490925) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L450)):*
This should be the number of bytes that make up the headers (ie. iterate over them and sum the length of each name and raw value)

---

*[components/devtools/actors/network_event.rs, line 458 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rtyW7iKN6fx3SPO30:-K4rtyW8ZU7mO9SNqGPL:-2015353911) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L458)):*
`match *self.request.headers.as_ref() {`

---

*[components/devtools/actors/network_event.rs, line 459 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4ru3HWR93jdCuCqrCk:-K4ru3HWR93jdCuCqrCl:-1054584143) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L459)):*
We can assign `cookies_size` here instead of using the `cookie` variable.

---

*[components/devtools/lib.rs, line 104 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4k-dKqIKQkyK7Bz-LL:-K4k-dKqIKQkyK7Bz-LM:-1461120978) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/lib.rs#L104)):*
nit: remove this blank line.

---

*[components/devtools/lib.rs, line 407 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4ruM1Tq18gmbRYXpmx:-K4ruM1Tq18gmbRYXpmy:-1607038494) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/lib.rs#L407)):*
nit: these lines should be indented 4 past the `let` on the previous line.

---

*[components/devtools/lib.rs, line 411 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4ruR_W1yq8shDlEIUq:-K4ruR_W1yq8shDlEIUr:180201478) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/lib.rs#L411)):*
nit: this should be indented to match the original `let`.

---

*[components/devtools/lib.rs, line 485 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rv6S16IJz4pTd0qtD:-K4rv6S16IJz4pTd0qtE:-1492696153) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/lib.rs#L485)):*
This comment can be removed now.

---

*[components/net/http_loader.rs, line 174 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rvD7CxmXwe5ScY5CQ:-K4rvD7CxmXwe5ScY5CR:1681114839) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/net/http_loader.rs#L174)):*
nit: This line can be removed.

---

*[components/net/http_loader.rs, line 187 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rvESAlytX_4V0OqPk:-K4rvESAlytX_4V0OqPl:1146973251) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/net/http_loader.rs#L187)):*
nit: These lines can be removed.

---

*[components/net/http_loader.rs, line 641 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rvyh6ieL9eaz_vLwh:-K4rvyh6ieL9eaz_vLwi:-946160310) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/net/http_loader.rs#L641)):*
We can inline `time::now()` here.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845#-:-K4rw2KgoxFa6Q7LI42_:-1706935459)*

**Comment by jdm** [comment]
Almost there! There are a bunch of review comments that weren't addressed, and the unit tests don't build yet - `./mach test-unit -p net` should let you see those.
-S-awaiting-review -S-fails-tidy +S-needs-code-changes 

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 21 unresolved discussions, some commit checks failed.

---

*[components/devtools/actors/network_event.rs, line 362 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jzghfaVaW8WndyAmA:-K5YtM9JGVOilYI2cLZH:2136711335) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L362)):*
This comment isn't necessary any more.

---

*[components/devtools/actors/network_event.rs, line 345 \[r2\]](https://reviewable.io:443/reviews/servo/servo/8845#-K5YtOj2u52bTRytZYFD:-K5YtOj2u52bTRytZYFE:-1105809266) ([raw file](https://github.com/servo/servo/blob/8df3373e8604a64b20de060863256e31e07e462c/components/devtools/actors/network_event.rs#L345)):*
These formats aren't identical - the result here is "04/22/2015 08:47:08". Let's do this instead: `format!("{}", self.request.startedDateTime.rfc3339())`

---

*[components/devtools/actors/network_event.rs, line 356 \[r2\]](https://reviewable.io:443/reviews/servo/servo/8845#-K5Yugjy2EWGpl3F5bX7:-K5Yugjy2EWGpl3F5bX8:-1080523218) ([raw file](https://github.com/servo/servo/blob/8df3373e8604a64b20de060863256e31e07e462c/components/devtools/actors/network_event.rs#L356)):*
This is equivalent to `let hSize = hSizeOption.unwrap_or(0);`

---

*[components/devtools/actors/network_event.rs, line 380 \[r2\]](https://reviewable.io:443/reviews/servo/servo/8845#-K5YuE9WWEuFSrZkcxyU:-K5YuE9WWEuFSrZkcxyV:-854587615) ([raw file](https://github.com/servo/servo/blob/8df3373e8604a64b20de060863256e31e07e462c/components/devtools/actors/network_event.rs#L380)):*
nit: unindent this by 4 spaces

---

*[components/devtools/actors/network_event.rs, line 398 \[r2\]](https://reviewable.io:443/reviews/servo/servo/8845#-K5YuB3rcM-OWq9waKse:-K5YuB3rcM-OWq9waKsf:-854587615) ([raw file](https://github.com/servo/servo/blob/8df3373e8604a64b20de060863256e31e07e462c/components/devtools/actors/network_event.rs#L398)):*
nit: unindent this by 4 spaces

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845#-:-K5YwJSvrS5Om_AeaH0T:-128621979)*

**Comment by jdm** [comment]
Still some unaddressed comments - are they confusing or causing problems?
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845#-:-K5bOZHelrQQBVlAxj65:123041506)*

**Comment by akumar21NCSU** [comment]
Review status: 4 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[components/devtools/actors/network_event.rs, line 175 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxVA5ki9gd-k4tK0J:-K5l_IVl3fTDon5gggmz:-1392127868) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L175)):*
Hi. I could not get this to work. It throws compilation error when I try what you suggested and I also tried - &*self.request_headers

---

*[components/devtools/actors/network_event.rs, line 179 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxfcR7iIIxPcx99cE:-K5mOuUS-sgNbi-KAje6:264576366) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L179)):*
Hi. This threw compilation error so I have not made this change.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845)*

**Comment by akumar21NCSU** [comment]
Review status: 4 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[components/devtools/actors/network_event.rs, line 385 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4k-1ak2IKv6nGFeQmt:-K5mYMhzWA4VdaHdk1HE:119690475) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L385)):*
This didn't compile. Error - expected `&hyper::http::RawStatus`,
    found `hyper::http::RawStatus` .  Change looks simple but with the suggested syntax I could not figure out how to git rid of this error.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845)*

**Comment by KiChjang** [comment]
Review status: 4 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[components/devtools/actors/network_event.rs, line 175 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxVA5ki9gd-k4tK0J:-K5mXV7vQgy_3DOVMH4a:-472523579) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L175)):*
What about `for item in self.request.headers.iter() {` and then removing the `if let Some(item) = headersIter.next() {` block altogether?

---

*[components/devtools/actors/network_event.rs, line 179 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxfcR7iIIxPcx99cE:-K5mZa0rCuIpoMSMd0m4:-59327607) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L179)):*
According to rust folks, `+=` is controlled by the AddAssign trait which isn't stable yet, so you can leave the code as-is.

---

*[components/devtools/actors/network_event.rs, line 385 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4k-1ak2IKv6nGFeQmt:-K5mb559Hj1z7ulSSY8i:676525275) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L385)):*
Try 
```Rust
let (status_code, status_message) =
    self.response.status.as_ref().map(|&RawStatus(ref code, ref text)| (*code, text.clone().into_owned())).unwrap_or((0, "".to_owned()));
```

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845)*

**Comment by akumar21NCSU** [comment]
Review status: 4 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[components/devtools/actors/network_event.rs, line 175 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jxVA5ki9gd-k4tK0J:-K5nAp7XOoyOXJqlpbTd:-543760066) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L175)):*
That worked. Thanks! 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845)*

**Comment by akumar21NCSU** [comment]
Review status: 4 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[components/devtools/actors/network_event.rs, line 385 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4k-1ak2IKv6nGFeQmt:-K5nBEYurGz4jQ4nZ3kw:-1834867277) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L385)):*
Thanks!

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845)*

**Comment by jdm** [comment]
Almost there! Thanks for addressing these!
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[components/devtools/actors/network_event.rs, line 237 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4jywWfAx64VLmBcHYH:-K5peOOOzCS9QyPVREth:1651871219) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L237)):*
This suggestion relies on the previous one for the `headers` variable, and should replace both the `for` loop and the `if let` right below it.

---

*[components/devtools/actors/network_event.rs, line 450 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rtJ1Fl9Vk_-wBlXB4:-K5perYhGjIr5Ua1AQ4G:-1719608117) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L450)):*
Could we at least document the algorithm for calculating headersSize in the TODO?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845#-:-K5pf8VmrMSxNwGenT2-:771708580)*

**Comment by akumar21NCSU** [comment]
Review status: 4 of 5 files reviewed at latest revision, 8 unresolved discussions.

---

*[components/devtools/actors/network_event.rs, line 450 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8845#-K4rtJ1Fl9Vk_-wBlXB4:-K5qX1oE7w6WzHh5jD_N:-499146009) ([raw file](https://github.com/servo/servo/blob/55bdce268f0d2b8f3fad82418e314345c0b52e69/components/devtools/actors/network_event.rs#L450)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845)*

**Comment by jdm** [comment]
Thanks! I'll squash these into one commit and merge them.
-S-awaiting-review +S-needs-squash 

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8845#-:-K5qaDcD3I3EzKIewWRj:1919594969)*

**Comment by jdm** [comment]
Ah, I see it's one commit already. In that case, could you modify the commit message to use the following instead: "Support responseCookies, responseContent, responseHeaders, requestCookies, getResponseHeaders, getResponseContent, getRequestPostData, getRequestCookies, getResponseCookies, getEventTimings and getSecurityInfo message for network devtools."

**Comment by akumar21NCSU** [comment]
Commit message changed.

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

File: components/devtools/actors/network_event.rs [hunk 1]
```diff
@@ -79,29 +167,124 @@ impl Actor for NetworkEventActor {
...
             "getRequestCookies" => {
-                ActorMessageStatus::Ignored
+                let mut cookies = Vec::new();
+                    if let Some(req_cookies) = self.request.headers.get_raw("Cookie") {
+                        for cookie in &*req_cookies {
+                            if let Ok(cookie_value) = String::from_utf8(cookie.clone()) {
+                                cookies = cookie_value.into_bytes();
+                            }
+                        }
+                }
```
Comment (KiChjang) [inline components/devtools/actors/network_event.rs]: The last `}` is indented wrongly. Can we fix it?

File: components/devtools/actors/network_event.rs [hunk 2]
```diff
@@ -79,29 +167,124 @@ impl Actor for NetworkEventActor {
...
+                        rawHeadersString = rawHeadersString + name + ":" + &value + "\r\n";
+                    }
+                    let msg = GetResponseHeadersReply {
+                        from: self.name(),
+                        headers: headerNames,
+                        headerSize: headersSize,
+                        rawHeaders: rawHeadersString,
+                    };
+                    stream.write_json_packet(&msg);
+                    }
```
Comment (KiChjang) [inline components/devtools/actors/network_event.rs]: Same here, the last `}` is indented wrongly.

**Comment by akumar21NCSU** [comment]
Hi. Do you mean the closing bracket for 'requestCookies' ? I see that it is in line(same column) with the starting column of this block. Should that change?

File: components/devtools/actors/network_event.rs (hunk 1 above)
Comment (jdm) [inline components/devtools/actors/network_event.rs]: It's actually the preceding block (starting with the `if let`) that is over-indented by four spaces.

**Comment by jdm** [comment]
See my response to the comment in question.

File: components/devtools/actors/network_event.rs (hunk 1 above)
Comment (akumar21NCSU) [inline components/devtools/actors/network_event.rs]: oh okay. It's fixed now.

**Pull request merged**